src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,41 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
+  const options = ['all', 'active', 'completed'];
+
+  const handleChangeStaus = (value: Status) => {
+    dispatch(
+      filterActions.setStatus(value),
+    );
+  };
+
+  const handleChangeQuery = (value = '') => {
+    dispatch(
+      filterActions.setQuery(value),
+    );
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={(e) => handleChangeStaus(e.target.value as Status)}
```

Comment:
In 'handleChangeStatus' you are casting 'e.target.value' to 'Status'. It's risky because the value might not belong to the 'Status' enum. Consider validating the value before casting it.